pkg/imrpc/proxy.proto

```diff
@@ -0,0 +1,344 @@
+syntax="proto3";
+
+package imrpc;
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/timestamp.proto";
+import "github.com/longhorn/longhorn-engine/proto/ptypes/controller.proto";
+import "github.com/longhorn/longhorn-engine/proto/ptypes/syncagent.proto";
+
+service ProxyService {
+	rpc Ping(google.protobuf.Empty) returns (ProxyACK);
+
+	rpc ServerVersionGet(ProxyDirect) returns (EngineServerVersionResponse);
+
+	rpc VolumeGet(ProxyDirect) returns (EngineVolumeGetResponse);
+	rpc VolumeExpand(EngineVolumeExpandRequest) returns (ProxyACK);
+	rpc VolumeFrontendStart(EngineVolumeFrontendStartRequest) returns (ProxyACK);
+	rpc VolumeFrontendShutdown(ProxyDirect) returns (ProxyACK);
+
+	rpc VolumeSnapshot(EngineVolumeSnapshotRequest) returns (EngineVolumeSnapshotResponse);
+	rpc SnapshotList(ProxyDirect) returns (EngineSnapshotListResponse);
+	rpc SnapshotRevert(EngineSnapshotRevertRequest) returns (ProxyACK);
+	rpc SnapshotPurge(EngineSnapshotPurgeRequest) returns (ProxyACK);
+	rpc SnapshotPurgeStatus(ProxyDirect) returns (EngineSnapshotPurgeStatusResponse);
+	rpc SnapshotClone(EngineSnapshotCloneRequest) returns (ProxyACK);
+	rpc SnapshotCloneStatus(ProxyDirect) returns (EngineSnapshotCloneStatusResponse);
+	rpc SnapshotRemove(EngineSnapshotRemoveRequest) returns (ProxyACK);
+
+	rpc SnapshotBackup(EngineSnapshotBackupRequest) returns (EngineSnapshotBackupResponse);
+	rpc SnapshotBackupStatus(EngineSnapshotBackupStatusRequest) returns (EngineSnapshotBackupStatusResponse);
+	rpc BackupRestore(EngineBackupRestoreRequest) returns (EngineBackupRestoreResponse);
+	rpc BackupRestoreStatus(ProxyDirect) returns (EngineBackupRestoreStatusResponse);
+	rpc BackupVolumeList(EngineBackupVolumeListRequest) returns (EngineBackupVolumeListResponse);
+	rpc BackupVolumeGet(EngineBackupVolumeGetRequest) returns (EngineBackupVolumeGetResponse);
+	rpc BackupGet(EngineBackupGetRequest) returns (EngineBackupGetResponse);
+	rpc BackupConfigMetaGet(EngineBackupConfigMetaGetRequest) returns (EngineBackupConfigMetaGetResponse);
+	rpc BackupRemove(EngineBackupRemoveRequest) returns (ProxyACK);
+
+	rpc ReplicaAdd(EngineReplicaAddRequest) returns (ProxyACK);
+	rpc ReplicaList(ProxyDirect) returns (EngineReplicaListResponse);
+	rpc ReplicaRebuildingStatus(ProxyDirect) returns (EngineReplicaRebuildStatusResponse);
+	rpc ReplicaVerifyRebuild(EngineReplicaVerifyRebuildRequest) returns (ProxyACK);
+	rpc ReplicaRemove(EngineReplicaRemoveRequest) returns (ProxyACK);
+}
+
+message ProxyDirect{
+	string to_controller = 1;
```

Comment:
Can we just rename to `address` given the message is `ProxyEngine`?

Reply:
Updated.